WallsAndHoles/xmltool.cpp

```diff
@@ -0,0 +1,283 @@
+#include "xmltool.h"
+#include <QDebug>
+#include <QMessageBox>
+
+SharedTileMap XMLTool::openTileMap(QString tileMapPath){
+    //load the file
+    QFile file(tileMapPath);
+    if (!file.exists() || !file.open(QFile::ReadOnly | QFile::Text)) {
+        qDebug() << "Fail to open file:"+tileMapPath;
+        return nullptr;
+    }
+    QXmlStreamReader xmlReader(file.readAll());
+    SharedTileMap tileMap;
+    //Parse the XML until we reach end of it
+    while(!xmlReader.atEnd() && !xmlReader.hasError()) {
+        // Read next element
+        QXmlStreamReader::TokenType token = xmlReader.readNext();
+        //If token is just StartDocument - go to next
+        if(token == QXmlStreamReader::StartDocument) {
+                continue;
+        }
+        //If token is StartElement - read it
+        if(token == QXmlStreamReader::StartElement) {
+            if(xmlReader.name() == "TileMap") {
+                QSize size;
+                foreach(const QXmlStreamAttribute &attr, xmlReader.attributes()) {
+                    if (attr.name().toString() == QString("width")) {
```

Comment:
should be able to write just `attr.name() == "width"` for this and all following cases?

Reply:
TileMap is deemed to have both width and height, but if for reason it didn't, you end up create a  tilemap with 0 width/height, I think it's a reasonable behavior.